### PR TITLE
Add associations

### PR DESCRIPTION
**Pull request by ruioliveira02**
An association is when a user ask to be a part of an organization. That request can be approved or dismissed by an admin of the org.

### CODE REVIEW

File: lib/atomic_web/live/association_live/form_component.ex [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    )
+  end
+
+  defp update_association(socket, :edit, association_params) do
+    association = socket.assigns.association
+    current_user = association_params["current_user"]
+    number = association_params["number"]
+
+    if association_params["accepted"] == "true" do
+      if not number do
```
Comment (MarioRodrigues10) [inline lib/atomic_web/live/association_live/form_component.ex:43]:
```suggestion
      if is_nil(number) do
```

File: lib/atomic_web/live/association_live/index.html.heex [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        <td>
+          <%= link to: Routes.association_show_path(@socket, :show, association.organization_id, association) do %>
+            Show
+          <% end %>
+        </td>
+      </tr>
+    <% end %>
+  </table>
+
+  <table :class="[toggle ? 'hidden' : 'block']">
```
Comment (MarioRodrigues10) [inline lib/atomic_web/live/association_live/index.html.heex:38]: It should be possible to always edit an association.

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

File: lib/atomic/organizations.ex [hunk 3]
```diff
@@ -4,8 +4,10 @@ defmodule Atomic.Organizations do
   """
 
   import Ecto.Query, warn: false
+  alias Jason.Encoder.Ecto.Association
```
Comment (MarioRodrigues10) [inline lib/atomic/organizations.ex:7]: Remove this line pleas

File: lib/atomic_web/live/organization_live/show.ex [hunk 4]
```diff
@@ -16,6 +16,26 @@ defmodule AtomicWeb.OrganizationLive.Show do
      |> assign(:organization, Organizations.get_organization!(id, [:departments]))}
   end
 
+  @impl true
+  def handle_event("sign_up", _params, socket) do
```
Comment (MarioRodrigues10) [inline lib/atomic_web/live/organization_live/show.ex:20]:
```suggestion
  def handle_event("sign_up", _payload, socket) do
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

**Review by MarioRodrigues10 — APPROVED** [review]
:heart: :orange_heart:

File: lib/atomic/accounts/user.ex [hunk 5]
```diff
@@ -18,6 +19,8 @@ defmodule Atomic.Accounts.User do
 
     has_many :enrollments, Enrollment
 
+    many_to_many :organizations, Organization, join_through: Association
```
Comment (feliciofilipe) [inline lib/atomic/accounts/user.ex:22]:
```suggestion
    many_to_many :organizations, Organization, join_through: Member
```

File: lib/atomic/accounts/user.ex [hunk 6]
```diff
@@ -5,6 +5,7 @@ defmodule Atomic.Accounts.User do
   use Atomic.Schema
 
   alias Atomic.Activities.Enrollment
+  alias Atomic.Organizations.{Association, Organization}
```
Comment (feliciofilipe) [inline lib/atomic/accounts/user.ex:8]:
```suggestion
  alias Atomic.Organizations.{Member, Organization}
```

**Review by feliciofilipe — CHANGES_REQUESTED** [review]
The best term for this use case is `Member`

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

File: lib/atomic_web/live/organization_live/show.html.heex [hunk 7]
```diff
@@ -23,5 +23,7 @@
   </li>
 </ul>
 
+<button phx-click="follow">Follow</button>
```
Comment (MarioRodrigues10) [inline lib/atomic_web/live/organization_live/show.html.heex:26]: It returns an error, when pressed twice

File: lib/atomic_web/router.ex [hunk 8]
```diff
@@ -55,6 +55,11 @@ defmodule AtomicWeb.Router do
       live "/organizations/:id/edit", OrganizationLive.Index, :edit
       live "/organizations/:id", OrganizationLive.Show, :show
       live "/organizations/:id/show/edit", OrganizationLive.Show, :edit
+
+      live "/memberships/:org", MembershipLive.Index, :index
+      live "/memberships/:org/new", MembershipLive.New, :new
```
Comment (MarioRodrigues10) [inline lib/atomic_web/router.ex:60]: User should show email or name and not id

File: lib/atomic_web/live/organization_live/show.html.heex (hunk 7 above)
Comment (MarioRodrigues10) [reply lib/atomic_web/live/organization_live/show.html.heex:26]: Actually this button should disappear if they already follow or if there's a pending invite

Comment (MarioRodrigues10) [reply lib/atomic_web/live/organization_live/show.html.heex:26]: Actually this button should disappear if they already follow or if there's a pending invite

**Pull request merged**